yamllint/rules/quoted_strings.py

```diff
@@ -259,9 +269,16 @@ def check(conf, token, prev, next, nextnext, context):
 
     elif conf['required'] == 'only-when-needed':
 
+        # Quotes needed on strings containing flow tokens
+        if context['flow_cnt'] > 0 and \
+           {',', '[', ']', '{', '}'} & set(token.value):
+            has_flow_token = True
+        else:
+            has_flow_token = False
```

Comment:
I suggest this code, which might be more Pythonic:
```suggestion
        has_flow_token = (context['flow_cnt'] > 0 and
                          token.value in (',', '[', ']', '{', '}'))
```

Reply:
"token.value in (',', '[', ']', '{', '}'))" doesn't work...